src/features/lightspeed/inlineSuggestions.ts

```diff
@@ -835,5 +823,54 @@ export async function inlineSuggestionUserActionHandler(
     inlineSuggestionFeedbackPayload,
     lightSpeedManager.orgTelemetryOptOut
   );
+  resetSuggestionData();
+}
+
+function inlineSuggestionPending(): boolean {
+  if (vscode.window.activeTextEditor?.document.languageId !== "ansible") {
+    return false;
+  }
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return false;
+  }
+  if (!inlineSuggestionData["suggestionId"]) {
+    return false;
+  }
+  return true;
+}
+
+function resetSuggestionData(): void {
   inlineSuggestionData = {};
+  insertTexts = [];
+}
+
+export async function textDocumentChangeHandler(
+  e: vscode.TextDocumentChangeEvent
+) {
+  // If the user accepted a suggestion on the widget, ansible.lightspeed.inlineSuggest.accept
+  // command is not sent. This method checks if a text change that matches to the current
+  // suggestion was found. If such a change was detected, we assume that the user accepted
+  // the suggestion on the widget.
+  if (
+    inlineSuggestionPending() &&
+    insertTexts &&
+    e.document.languageId === "ansible" &&
+    e.contentChanges.length > 0
+  ) {
+    const suggestionId = inlineSuggestionData["suggestionId"];
+
+    e.contentChanges.forEach(async (c) => {
+      if (c.text === insertTexts[0]) {
```

Comment:
Does this work for both single task and multi-task correctly? iirc there is some extra handling done for single task suggestion.

Reply:
As far as I tested it, it seems working both for single & multi-task... Do you remember what was the extra handling for single task suggestion?

Reply:
For single task we add the task name to the accepted suggestion here https://github.com/ansible/vscode-ansible/blob/24af91bde2cbc1293b38dcff11018c8180f32f33/src/features/lightspeed/inlineSuggestions.ts#L423

Reply:
If I read the code correctly, I think the additional task name added to `contentMatchesForSuggestion` is used only for content matches, not stored in `inlineSuggestionUserActionItems`. I think we can depend on the value in `insertTexts` for string comparison in text change handler. Or the `contentMatchesForSuggestion` would somehow influent text changes made by inline suggestion?